Engine/DataFeeds/SubscriptionDataReader.cs

```diff
@@ -372,7 +372,7 @@ public bool MoveNext()
 
                     // if we move past our current 'date' then we need to do daily things, such
                     // as updating factors and symbol mapping
-                    if (instance.EndTime.Date > _tradeableDates.Current)
+                    if (instance.EndTime.ConvertTo(_config.ExchangeTimeZone, _config.DataTimeZone).Date > _tradeableDates.Current)
```

Comment:
Nice! This could actually mean we were losing some data points too, right? Depending on the diff between the data and the exchange time zone, we might have created a new enumerator before time

Reply:
Yes, exactly 👍